src/main/java/classes/Event.java

```diff
@@ -0,0 +1,62 @@
+package classes;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+
+public class Event extends Task {
+    protected String newStartDate, newEndDate, newStartTime, newEndTime,
+            oldStartDate, oldEndDate, oldStartTime, oldEndTime;
+
+    protected LocalTime startTime, endTime;
+
+    protected LocalDate startDate, endDate;
+
+    public Event(String taskInfo, String stringStartDate, String stringEndDate, String stringStartTime
+            , String stringEndTime) {
+        super(taskInfo);
+        this.oldStartDate = stringStartDate;
+        this.oldEndDate = stringEndDate;
+        this.oldStartTime = " " + stringStartTime;
+        this.oldEndTime = " " + stringEndTime;
+        this.startDate = LocalDate.parse(stringStartDate);
+        this.newStartDate = DateTimeFormatter.ofLocalizedDate(FormatStyle.LONG)
+                .format(startDate);
+        this.endDate = LocalDate.parse(stringEndDate);
+        this.newEndDate = DateTimeFormatter.ofLocalizedDate(FormatStyle.LONG)
+                .format(endDate);
```

Comment:
The constructor for this and other Task classes is quite difficult to understand, is there any way of making it clearer?

Reply:
Alright, thanks for the advice! I will review through this section and release an improved version!